app/models/condition_set.rb

```diff
@@ -4,7 +4,7 @@ class ConditionSet < ApplicationRecord
   include Reviewable
 
   belongs_to :planning_application
-  has_many :conditions, extend: ConditionsExtension
+  has_many :conditions, extend: ConditionsExtension # rubocop:disable Rails/HasManyOrHasOneDependent
```

Comment:
This should be `dependent: :destroy` as the planning application has `dependent: :destroy` on the `:condition_set`